### PR TITLE
Added Documentation for Procedural Sky

### PR DESCRIPTION
**Pull request by clayjohn**
Let me know if anything needs to be fixed or improved!

### CODE REVIEW

File: doc/classes/ProceduralSky.xml [hunk 1]
```diff
@@ -12,36 +15,52 @@
 	</methods>
 	<members>
 		<member name="ground_bottom_color" type="Color" setter="set_ground_bottom_color" getter="get_ground_bottom_color">
+			Color of the ground at the bottom
```
Comment (mhilbrunner) [inline doc/classes/ProceduralSky.xml]: Missing '.'

File: doc/classes/ProceduralSky.xml [hunk 2]
```diff
@@ -1,8 +1,11 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <class name="ProceduralSky" inherits="Sky" category="Core" version="3.1">
 	<brief_description>
+		Type of [Sky] that is generated procedurally based on user input parameters.
 	</brief_description>
 	<description>
+		The ProceduralSky provides a means for users to create an effective background quickly for prototyping or to create a background for a custom scene. It has three main components: the sun, the sky, and the ground. The sky and ground are very similar, they are defined by a color at the horizon, another color, and finally an easing curve to interpolate between these two colors. Similarly the sun is described by a position in the sky, a color, and an easing curve. However, the sun also defines a minimum and maximum angle, these two values define at what distance the easing curve begins and ends from the sun, and thus end up defining the size of the sun in the sky.
```
Comment (mhilbrunner) [inline doc/classes/ProceduralSky.xml]: IMO, shorten first two sentences to "ProceduralSky provides a way to create an effective background quickly by defining procedural parameters for the sun, the sky and the ground."

File: doc/classes/ProceduralSky.xml [hunk 3]
```diff
@@ -12,36 +15,52 @@
...
+			Color of the sun.
 		</member>
 		<member name="sun_curve" type="float" setter="set_sun_curve" getter="get_sun_curve">
+			How quickly the sun fades away between [member sun_angle_min] and [member sun_angle_max]
 		</member>
 		<member name="sun_energy" type="float" setter="set_sun_energy" getter="get_sun_energy">
+			Amount of energy contribution from the sun.
 		</member>
 		<member name="sun_latitude" type="float" setter="set_sun_latitude" getter="get_sun_latitude">
+			Angle of the sun in the sky vertically.
```
Comment (mhilbrunner) [inline doc/classes/ProceduralSky.xml]: Vertical angle of the sun? Not sure.

File: doc/classes/ProceduralSky.xml [hunk 4]
```diff
@@ -12,36 +15,52 @@
...
+			How quickly the sun fades away between [member sun_angle_min] and [member sun_angle_max]
 		</member>
 		<member name="sun_energy" type="float" setter="set_sun_energy" getter="get_sun_energy">
+			Amount of energy contribution from the sun.
 		</member>
 		<member name="sun_latitude" type="float" setter="set_sun_latitude" getter="get_sun_latitude">
+			Angle of the sun in the sky vertically. 
 		</member>
 		<member name="sun_longitude" type="float" setter="set_sun_longitude" getter="get_sun_longitude">
+			Angle of the sun in the sky horizontally.
```
Comment (mhilbrunner) [inline doc/classes/ProceduralSky.xml]: Horizontal angle of the sun? Not sure.

File: doc/classes/ProceduralSky.xml (hunk 2 above)
Comment (clayjohn) [reply doc/classes/ProceduralSky.xml]: I like it. Thanks for the feedback!

File: doc/classes/ProceduralSky.xml (hunk 3 above)
Comment (clayjohn) [reply doc/classes/ProceduralSky.xml]: What about 'vertical position of the sun described as an angle above the horizon'?

**Comment by clayjohn** [comment]
Should be good to go now

**Comment by mhilbrunner** [comment]
Thank you!

**Comment by hpvb** [comment]
Cherry picked into 3.0.3

**Pull request merged**